met-web/src/components/survey/submit/SurveySubmitWrapped.tsx

```diff
@@ -30,17 +30,17 @@ const SurveySubmitWrapped = () => {
                 </Grid>
                 <Grid item xs={12}>
                     <MetPaper elevation={2}>
-                        <ConditionalComponent condition={isTokenValid && savedSurvey.engagement}>
+                        <ConditionalComponent condition={isTokenValid && Boolean(savedSurvey.engagement)}>
                             <SurveyForm
                                 handleClose={() => {
-                                    navigate(`/engagements/${savedSurvey.engagement.id}/view`);
+                                    navigate(`/engagements/${savedSurvey.engagement?.id}/view`);
                                 }}
                             />
                         </ConditionalComponent>
                         <InvalidTokenModal
-                            open={!isTokenValid && savedSurvey.engagement}
+                            open={!isTokenValid && Boolean(savedSurvey.engagement)}
```

Comment:
what does the boolean do here?

Reply:
This change really isn't part of my task. I had a code strictness error in the Web build action and it complained that savedSurvey.engagement could be null or undefined and there's a && before it meaning it is expecting it to be Boolean. So when you wrap Boolean() around any object it really checks if it's truthy or Falsey. and spits out a boolean So null, undefined will be false and anything else true. With this the error went away